### PR TITLE
feat(cdc): Apply FINAL clause in CDC tables

### PR DESCRIPTION
**Pull request by nikhars**
Since data in CDC tables are mutable and these tables are fairly smaller
compared to the other tables, we can apply the FINAL clause for queries
to these tables.

Also removed the final member of the Table class from equality tests
since logically any 2 tables with or without the final clause should
be the same.

### CODE REVIEW

File: snuba/query/data_source/simple.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ class Table(SimpleDataSource):
 
     table_name: str
     schema: ColumnSet
-    final: bool = False
+    final: bool = field(default=False, compare=False)
```
Comment (evanh) [inline snuba/query/data_source/simple.py:48]: Why did you need this?

**Comment by lynnagara** [comment]
I was a bit surprised to see all the existing tests pass even though the generated Clickhouse SQL statement should have changed to include the final keyword. For example these tests https://github.com/getsentry/snuba/blob/master/tests/clickhouse/test_query_format.py join errors with grouped message and assert on the generated SQL. Do you know where there's no FINAL there?

File: snuba/query/data_source/simple.py (hunk 1 above)
Comment (nikhars) [reply snuba/query/data_source/simple.py:48]: When running tests, there was this single test which failed. https://github.com/getsentry/snuba/blob/master/tests/pipeline/test_composite_planner.py#L466
The difference was the final member field. 

I think the final field should not matter in equality tests since its more a matter of applying optimization. Let me know if you think we should to consider it for equality tests and I can then fix the tests

**Comment by nikhars** [comment]
> I was a bit surprised to see all the existing tests pass even though the generated Clickhouse SQL statement should have changed to include the final keyword. For example these tests https://github.com/getsentry/snuba/blob/master/tests/clickhouse/test_query_format.py join errors with grouped message and assert on the generated SQL. Do you know why there's no FINAL there?

Let me check

**Comment by nikhars** [comment]
> I was a bit surprised to see all the existing tests pass even though the generated Clickhouse SQL statement should have changed to include the final keyword. For example these tests https://github.com/getsentry/snuba/blob/master/tests/clickhouse/test_query_format.py join errors with grouped message and assert on the generated SQL. Do you know why there's no FINAL there?

The tests don't run the processors on the query. They only format it. Hence there's no FINAL there.

File: snuba/query/data_source/simple.py (hunk 1 above)
Comment (lynnagara) [reply snuba/query/data_source/simple.py:48]: How about updating the processed query in the test_composite_planner to just have final=True like the processing pipeline will produce so this isn't required. https://github.com/getsentry/snuba/blob/master/tests/pipeline/test_composite_planner.py#L71

If we can avoid it I think we should avoid skipping the comparison on this field, as it causes two queries that are not the same to be (arguably incorrectly) reported as equal. I think it will make it harder to write tests that do check for the final behavior, which we may want to have. This code is not test specific so I also worry that it introduces the potential for incorrect comparisons at runtime.

**Review by evanh — APPROVED** [review]

**Review by lynnagara — APPROVED** [review]
Thanks
